### PR TITLE
Adding contracts for wrapped tokens

### PR DESCRIPTION
**Pull request by AymericNoel**

### CODE REVIEW

File: src/Trakx.Contracts/Wrapping/WrappedToken/WrappedTokenService.cs [hunk 1]
```diff
@@ -0,0 +1,595 @@
...
+        {
+            var receipt = await DeployContractAndWaitForReceiptAsync(web3, wrappedTokenDeployment, cancellationTokenSource);
+            return new WrappedTokenService(web3, receipt.ContractAddress);
+        }
+
+        protected Nethereum.Web3.Web3 Web3{ get; }
+
+        public ContractHandler ContractHandler { get; }
+
+        public WrappedTokenService(Nethereum.Web3.Web3 web3, string tokenSymbol)
```
Comment (AymericNoel) [inline src/Trakx.Contracts/Wrapping/WrappedToken/WrappedTokenService.cs:39]: I put the symbol of the token here to retrieve the correct address for the correct contract.

File: src/Trakx.Contracts/Wrapping/DeployedContractAddresses.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace Trakx.Contracts.Wrapping
+{
+    public static class DeployedContractAddresses
+    {
+        public static readonly string wxrp = "0xbbD8689321a543510a67e4a9F9D5e7B0C2de5257"; //rinkeby network
+
+        public static readonly ReadOnlyDictionary<string, string> AddressByName = new ReadOnlyDictionary<string, string>(
+            typeof(DeployedContractAddresses)
+                .GetFields(BindingFlags.Public | BindingFlags.Static | BindingFlags.FlattenHierarchy)
```
Comment (monsieurleberre) [inline src/Trakx.Contracts/Wrapping/DeployedContractAddresses.cs:16]: Where is this dodgy code copy pasted from 😉😜 ???
Should we do a method from the `.GetFields` to the `;`

File: src/Trakx.Contracts/Wrapping/WrappedToken/ContractDefinition/WrappedTokenDefinition.cs [hunk 3]
```diff
@@ -0,0 +1,577 @@
...
...6f6b656e20726563697069656e7420636f6e747261637420686173206e6f20696d706c656d656e74657220666f7220455243373737546f6b656e73526563697069656e7443616c6c657220646f6573206e6f742068617665207468652061646d696e20726f6c654552433737373a207472616e7366657220746f20746865207a65726f20616464726573734552433737373a2063616c6c6572206973206e6f7420616e206f70657261746f7220666f7220686f6c6465724552433737373a207472616e7366657220616d6f756e74206578636565647320616c6c6f77616e6365537065636966696564206164647265737320697320616c72656164792061206d696e7465722e4552433737373a207472616e736665722066726f6d20746865207a65726f20616464726573735370656369666965642061646472657373206973206e6f742061206275726e65722e43616c6c657220646f6573206e6f74206861766520746865206d696e74657220726f6c654552433737373a206275726e20616d6f756e7420657863656564732062616c616e63654552433737373a20617070726f766520746f20746865207a65726f2061646472657373a265627a7a723158202b59fd8a98f83d674d6188d8f259ab6996babfdbdb130cd7051f114c07467eea64736f6c63430005100032526f6c65733a206163636f756e7420697320746865207a65726f206164647265737343616c6c657220646f6573206e6f742068617665207468652061646d696e20726f6c65537065636966696564206164647265737320697320616c72656164792061206d696e7465722e";
```
Comment (monsieurleberre) [inline src/Trakx.Contracts/Wrapping/WrappedToken/ContractDefinition/WrappedTokenDefinition.cs:25]: I see a mistake around `...42061646472657...` can you double check? 😂

File: src/Trakx.Contracts/Wrapping/WrappedToken/WrappedTokenService.cs (hunk 1 above)
Comment (monsieurleberre) [reply src/Trakx.Contracts/Wrapping/WrappedToken/WrappedTokenService.cs:39]: OK cool, that should work 👍

**Pull request merged**